src/quest_manager/views.py

```diff
@@ -930,17 +931,20 @@ def drop(request, submission_id):
 
 @login_required
 def submission(request, submission_id=None, quest_id=None):
-    # sub = QuestSubmission.objects.get(id = submission_id)
-    sub = get_object_or_404(QuestSubmission, pk=submission_id)
-    if sub.user != request.user and not request.user.is_staff:
-        return redirect('quests:quests')
+
+    initial = {}
 
     if request.user.is_staff:
-        # Staff form has additional fields such as award granting.
-        main_comment_form = SubmissionFormStaff(request.POST or None)
+        qs = QuestSubmission.objects.all_completed()
+        sub = get_object_or_404(qs, pk=submission_id)
+        CustomSubmissionForm = SubmissionFormStaff
     else:
-        initial = {'comment_text': sub.draft_text}
-        main_comment_form = SubmissionForm(request.POST or None, initial=initial)
+        qs = QuestSubmission.objects.all_completed(request.user)
+        sub = get_object_or_404(qs, pk=submission_id)
+        initial['comment_text'] = sub.draft_text
+        CustomSubmissionForm = SubmissionForm
+
+    main_comment_form = CustomSubmissionForm(request.POST or None, initial=initial)
```

Comment:
Please add comments explaining these code changes.  It's not obvious to me what's going on here and I don't want to waste tons of time trying to parse the code.  More code comments everywhere please!

Reply:
I'll add comments to it but what I was trying to do here was that I'm pre-filtering quest submissions. If it's a staff then it should fetch all completed submissions. And it should also use the SubmissionFormStaff form that will be used for instantiating later.

I did that instead of having this condition

```python
if request.user.is_staff:
    main_comment_form = SubmissionFormStaff(request.POST or None)
else:
    main_comment_form = SubmissionForm(request.POST or None, initial=initial)
```